bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS4Handler.java

```diff
@@ -0,0 +1,821 @@
+        sendPacketEncrypted(outPacket);
+    }
+
+    /**
+     * Ask PS4 if the OSK is open so we can get and set text.
+     */
+    private void sendOSKStart() {
+        ByteBuffer outPacket = PS4PacketHandler.makeOSKStartPacket();
+        sendPacketEncrypted(outPacket);
+    }
+
+    /**
+     * Sets the entire OSK string on the PS4.
+     *
+     * @param text The text to set in the OSK.
+     */
+    private void setOSKText(String text) {
+        logger.debug("Sending osk text packet,\"{}\"", text);
+        ByteBuffer outPacket = PS4PacketHandler.makeOSKStringChangePacket(text);
+        sendPacketEncrypted(outPacket);
+    }
+
+    /**
+     * Tries to start an application on the PS4.
+     *
+     * @param applicationId The unique id for the application (CUSAxxxxx).
+     */
+    private void startApplication(String applicationId) {
+        ByteBuffer outPacket = PS4PacketHandler.makeApplicationPacket(applicationId);
+        sendPacketEncrypted(outPacket);
+    }
+
+    private void sendRemoteKey(int pushedKey) {
+        try {
+            SocketChannel channel = getConnection();
+
+            ByteBuffer outPacket = PS4PacketHandler.makeRemoteControlPacket(PS4_KEY_OPEN_RC);
+            sendPacketEncrypted(outPacket, channel);
+            Thread.sleep(MIN_SENDKEY_DELAY);
+
+            // Send remote key
+            outPacket = PS4PacketHandler.makeRemoteControlPacket(pushedKey);
+            sendPacketEncrypted(outPacket, channel);
+            Thread.sleep(MIN_HOLDKEY_DELAY);
+
+            outPacket = PS4PacketHandler.makeRemoteControlPacket(PS4_KEY_OFF);
+            sendPacketEncrypted(outPacket, channel);
+            Thread.sleep(MIN_SENDKEY_DELAY);
+
+            outPacket = PS4PacketHandler.makeRemoteControlPacket(PS4_KEY_CLOSE_RC);
+            sendPacketEncrypted(outPacket, channel);
+
+        } catch (InterruptedException e) {
+            logger.debug("RemoteKey interrupted: {}", e.getMessage());
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.info("RemoteKey exception: {}", e.getMessage());
+        }
+    }
+
+    private void parseSearchResponse(DatagramPacket packet) {
+        byte[] data = packet.getData();
+        String message = new String(data, StandardCharsets.UTF_8);
+        String applicationName = "";
+        String applicationId = "";
+
+        String[] ss = message.trim().split("\n");
+        for (String row : ss) {
+            int index = row.indexOf(':');
+            if (index == -1) {
+                OnOffType power = null;
+                if (row.contains("200")) {
+                    power = OnOffType.ON;
+                } else if (row.contains("620")) {
+                    power = OnOffType.OFF;
+                }
+                if (power != null) {
+                    if (!currentPower.equals(power)) {
```

Comment:
no need to check that. the framework takes care of duplicate events

Reply:
Fixed.